controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -123,6 +129,39 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, r.handleDeletion(ctx, o)
 	}
 
+	// keep track of the HCPVaultSecretsApp for the cleanup goroutine
+	if HCPVaultSecretsAppMap[req.NamespacedName.String()] == nil {
+		HCPVaultSecretsAppMap[req.NamespacedName.String()] = o
+	}
+
+	if !initializedOrphanedShadowSecretCleanup {
+		initializedOrphanedShadowSecretCleanup = true
+
+		go func() {
+			// check for cleanup every 1 hour
+			ticker := time.NewTicker(1 * time.Hour)
```

Comment:
I wasn't sure what an appropriate interval should be, so I just set it to run once every hour for now.